.github/workflows/ci.yml

```diff
@@ -185,11 +185,17 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Configure Developer Command Prompt for Microsoft Visual C++
-        # Using microsoft/setup-msbuild is not enough.
-        uses: ilammy/msvc-dev-cmd@v1
-        with:
-          arch: x64
+      - name: Locate MSBuild and add to PATH
+        shell: pwsh
+        run: |
+          $vswherePath = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+          $msbuildPath = & "$vswherePath" -latest -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe
+          if (-not $msbuildPath) {
+            throw "MSBuild not found"
```

Comment:
Why use an exception? Wouldn’t exiting with an error code suffice?

Reply:
I spose I'd ask the question the other way round, why not the exception? It would be an exceptional circumstance if MSBuild could not be found.

Reply:
No, this [doc](https://docs.github.com/en/actions/sharing-automations/creating-actions/setting-exit-codes-for-actions) just says use any exit code other than 0 (which throw does do anyway).

I've tested throw works here: https://github.com/m3dwards/bitcoin/actions/runs/15050358923/job/42303402427

All that said, I'm really not wedded to it and happy to change to a simple `exit 1` if you prefer.

Reply:
I have no preferences either.